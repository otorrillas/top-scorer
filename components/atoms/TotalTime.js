import React from 'react';
import PropTypes from 'prop-types';

const TotalTime = ({ time }) => {
  const minutes = Math.floor(time / 60);
  const seconds = time % 60;

  return (
    <>
      {minutes} minutes and {seconds} seconds
    </>
  );
};

TotalTime.propTypes = {
  time: PropTypes.number.isRequired
};

export default TotalTime;
