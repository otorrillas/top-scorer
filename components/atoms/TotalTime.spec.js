import React from 'react';
import renderer from 'react-test-renderer';

import TotalTime from './TotalTime';

describe('<TotalTime />', () => {
  const renderComponent = props => renderer.create(<TotalTime {...props} />);

  describe('@render', () => {
    it('default', () => {
      const tree = renderComponent({
        time: 450
      });
      expect(tree).toMatchSnapshot();
    });
  });
});
