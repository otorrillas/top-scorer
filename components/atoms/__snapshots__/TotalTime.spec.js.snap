// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalTime /> @render default 1`] = `
Array [
  "7",
  " minutes and ",
  "30",
  " seconds",
]
`;
